src/Aspire.Hosting/DistributedApplicationRunner.cs

```diff
@@ -5,18 +5,38 @@
 using Aspire.Hosting.Publishing;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
 
 namespace Aspire.Hosting;
 
-internal sealed class DistributedApplicationRunner(IHostApplicationLifetime lifetime, DistributedApplicationExecutionContext executionContext, DistributedApplicationModel model, IServiceProvider serviceProvider) : BackgroundService
+internal sealed class DistributedApplicationRunner(ILogger<DistributedApplicationRunner> logger, IHostApplicationLifetime lifetime, DistributedApplicationExecutionContext executionContext, DistributedApplicationModel model, IServiceProvider serviceProvider, IPublishingActivityProgressReporter activityReporter) : BackgroundService
 {
     protected override async Task ExecuteAsync(CancellationToken stoppingToken)
     {
         if (executionContext.IsPublishMode)
         {
-            var publisher = serviceProvider.GetRequiredKeyedService<IDistributedApplicationPublisher>(executionContext.PublisherName);
-            await publisher.PublishAsync(model, stoppingToken).ConfigureAwait(false);
-            lifetime.StopApplication();
+            var publishingActivity = await activityReporter.CreateActivityAsync(
+                "docker-compose",
+                "Docker Compose artifacts",
+                isPrimary: true,
+                stoppingToken).ConfigureAwait(false);
```

Comment:
This  doesn't look right. Why does it say docker compose.

Reply:
Generalized.